### ISSUE TITLE
1.0.0

### ISSUE THREAD

**Issue by UncleDave** [aberezkin/ng2-image-upload#102]
Release checklist

- [x] API refactor
- [ ] Disable select button - #33
- [ ] Prefix internal CSS & allow custom css - #93
- [x] beforeUpload function - #59

@aberezkin please add anything I've missed for 1.0.0

e.g. we may want automated testing before 1.0.0?

**Comment by aberezkin** [comment]
Yeah, I think that automated tests are essential for 1.0.0. Partly because we want to submit for awesome angular components as I mentioned in #96. This will cause more users for our lib and therefore more contributors and more PRs.

**Comment by BenRacicot** [comment]
Hey guys, love this package! I'd love to help if I can. Perhaps at 1.0.0 rename to ng-image-upload? Also I was looking for client side resizing. My research has led me to believe that image uploading can be one of the heaviest bandwidth costs when you're paying per MB/GB like on AWS...

One solution is @bergben's https://github.com/bergben/ng2-img-max maybe he would want to combine forces?

Another bell or whistle would be client side image conversion to `.webp` this combined with resizing could reduce bandwidth exponentially. Anyways, I'd like to help!

**Comment by bergben** [comment]
Hi @BenRacicot, 

I think you are right about cutting bandwidth costs through client size image resizing.  `ng2-img-max` goal is to do exactly that, but there are of course cases where an image has to stay exactly the way the user uploads it. So I think with the introduced `beforeUpload` functionality (https://github.com/aberezkin/ng2-image-upload/pull/101) the users could do that themselves if they want to. Maybe even use https://github.com/bergben/ng2-img-tools to create thumbnails and upload them as well I think? :)

Regarding the `.webp` conversion: This would be a good idea, but be aware of the current browser support for webp: http://caniuse.com/#search=webp ... The users of the library would have to implement a way to convert it back in order to support all browsers...

Anyhow, feel free to use any of my packages or suggest changes! :) 


FYI: There are some issues with ng2-img-max, which I currently don't have time to resolve:

ng2-img-max uses [pica](https://github.com/nodeca/pica) under the hood, or to be more specific a fork of pica that I created for `ng2-pica`. See https://github.com/bergben/ng2-pica/issues/3 
Then there is the Alpha Channel issue, which also is related to `.webp`: https://github.com/bergben/ng2-img-max/issues/7 
Also the resizing does not seem to work for Safari currently, see https://github.com/bergben/ng2-img-tools/issues/14

**Comment by UncleDave** [comment]
I think we should get a pre-release on the go, a lot of good features have been added and there's no sense sitting on them.

Perhaps `1.0.0-rc1`

**Comment by hardikpthv** [comment]
Totally agree with @UncleDave.

**Comment by sabrio** [comment]
I agree too, we definitely need pre-release.

**Comment by aberezkin** [comment]
Yep, I agree sorry for that idle. Community is growing so fast :D
